api/src/main/java/org/openmrs/module/appframework/service/AppFrameworkServiceImpl.java

```diff
@@ -93,29 +97,17 @@ public AppFrameworkServiceImpl(AllAppTemplates allAppTemplates, AllAppDescriptor
 		this.javascriptEngine = new ScriptEngineManager().getEngineByName("JavaScript");
 		this.allUserApps = allUserApps;
 
-
-		// there is surely a cleaner way to define this utility function in the global scope
```

Comment:
This is my version of a "cleaner way".  This eliminates the global scope entirely, as it introduces unnecessary complexity for what I expect is negligible performance gain, and just keeps all of the various functions as their source strings, which are appended to the script that is executed each time.

This also moves the definition of these functions out of this very-hard-to-understand-and-maintain string concatenation and into javascript resource files that can benefit from syntax highlighting, code completion, formatting, and all of the other benefits that come with it.